tsdb/record/record.go

```diff
@@ -42,6 +43,48 @@ const (
 	Tombstones Type = 3
 	// Exemplars is used to match WAL records of type Exemplars.
 	Exemplars Type = 4
+	// Metadata is used to match WAL records of type Metadata.
+	Metadata Type = 5
+)
+
+// MetricType represents the type of a series.
+type MetricType uint8
+
+const (
+	UnknownMT      MetricType = 0
+	Counter        MetricType = 1
+	Gauge          MetricType = 2
+	Histogram      MetricType = 3
+	GaugeHistogram MetricType = 4
+	Summary        MetricType = 5
+	Info           MetricType = 6
+	Stateset       MetricType = 7
+)
```

Comment:
LGTM assuming these are the current metric types and how/where we want to encode them for source of truth. @codesome ?

Reply:
Could you provide some more context on this?

Reply:
I just happen to know that these are the types present in the client_golang.